packages/edit-site/src/hooks/commands/use-common-commands.js

```diff
@@ -288,6 +310,91 @@ function useGlobalStylesOpenRevisionsCommands() {
 		isEditorPage,
 		getCanvasMode,
 		setCanvasMode,
+		closeGeneralSidebar,
+	] );
+
+	return {
+		isLoading: false,
+		commands,
+	};
+}
+
+function useGlobalStylesStyleBookCommands() {
+	const {
+		openGeneralSidebar,
+		closeGeneralSidebar,
+		setEditorCanvasContainerView,
+		setCanvasMode,
+	} = unlock( useDispatch( editSiteStore ) );
+	const { getCanvasMode, getEditorCanvasContainerView } = unlock(
+		useSelect( editSiteStore )
+	);
+	const canvasContainerView = getEditorCanvasContainerView();
+	const { params } = useLocation();
+	const isMobileViewport = useViewportMatch( 'medium', '<' );
+	const isEditorPage = ! getIsListPage( params, isMobileViewport );
+	const history = useHistory();
+	const { isStyleBookOpened, isRevisionsStyleBookOpened, isRevisionsOpened } =
+		useSelect( () => {
+			return {
+				isStyleBookOpened: 'style-book' === canvasContainerView,
+				isRevisionsStyleBookOpened:
+					'global-styles-revisions:style-book' ===
+					canvasContainerView,
+				isRevisionsOpened:
+					'global-styles-revisions' === canvasContainerView,
+			};
+		}, [] );
+	const toggleStyleBook = () => {
+		if ( isRevisionsOpened ) {
+			setEditorCanvasContainerView(
+				'global-styles-revisions:style-book'
+			);
+			return;
+		}
+		if ( isRevisionsStyleBookOpened ) {
+			setEditorCanvasContainerView( 'global-styles-revisions' );
+			return;
+		}
+		if ( isStyleBookOpened ) {
+			closeGeneralSidebar( 'edit-site/global-styles' );
+		} else {
+			openGeneralSidebar( 'edit-site/global-styles' );
+		}
+		setEditorCanvasContainerView(
+			isStyleBookOpened ? undefined : 'style-book'
+		);
```

Comment:
To allow opening from the site editor "view" sidebar, I think we'd also need to account for the canvas view mode and set it to "edit" if it isn't.

Similar to how the styles revisions callback works:

```
					if ( isEditorPage && getCanvasMode() !== 'edit' ) {
						setCanvasMode( 'edit' );
					}
```

<img width="431" alt="Screenshot 2024-01-29 at 11 25 50 am" src="https://github.com/WordPress/gutenberg/assets/6458278/3a4924ca-424d-44e1-92cc-45b5460e6978">